### PR TITLE
Add test for helm charts

### PR DESCRIPTION
**Pull request by ehabshaaban**
- [x] generate kubernetes clientset
- [x] write clientConfig in `~/.kube/config`
- [x] ValidatePrometheusDeployment
- [x] ValidateGrafanaDeployment

### CODE REVIEW

File: test/infra_test.go [hunk 1]
```diff
@@ -1,126 +1,228 @@
...
+	ca, err := base64.StdEncoding.DecodeString(aws.StringValue(cluster.CertificateAuthority.Data))
+	if err != nil {
+		return nil, err
+	}
+
+	clientset, err := kubernetes.NewForConfig(
+		&rest.Config{
+			Host:        aws.StringValue(cluster.Endpoint),
+			BearerToken: tok.Token,
+			TLSClientConfig: rest.TLSClientConfig{
```
Comment (omarahm3) [inline test/infra_test.go:74]: im on mobile so i cant check, but looks like indentation is not right. please make sure that you run `go fmt` on whole file

File: .github/workflows/terraform-deploy.yml [hunk 2]
```diff
@@ -43,12 +44,12 @@ jobs:
       - name: Terraform init
         uses: ./.github/actions/setup
       
-      # - name: Terraform test
-      #   run: cd test && go test -v -timeout 2000s infra_test.go
-      #   continue-on-error: true
+      - name: Terraform test
+        run: cd test && go test -v -timeout 3000s infra_test.go
```
[the remark this answers is not in the document]
Comment (ehabshaaban) [reply .github/workflows/terraform-deploy.yml:48]: You're right. But'll choose not to do it for the moment.

File: test/infra_test.go [hunk 3]
```diff
@@ -1,126 +1,228 @@
...
-	var podName string
+// https://stackoverflow.com/questions/60547409/unable-to-obtain-kubeconfig-of-an-aws-eks-cluster-in-go-code/60573982#60573982
+func newClientset(cluster *eks.Cluster) (*kubernetes.Clientset, error) {
+	const (
+		RecommendedHomeDir  = ".kube"
+		RecommendedFileName = "config"
+	)
 
-	pods := k8s.ListPods(t, kubeOptions, v1.ListOptions{FieldSelector: "status.phase=Running"})
+	var RecommendedHomeFile = path.Join(homedir.HomeDir(), RecommendedHomeDir, RecommendedFileName)
```
[the remark this answers is not in the document]
Comment (ehabshaaban) [reply test/infra_test.go:48]: `RecommendedHomeDir`: is `.kube` directory
`RecommendedFileName`: is name of config file
`RecommendedHomeFile`: is machine instance home directory + `.kube` directory + name of config file
Where is the naming conflict is coming from?

File: test/infra_test.go [hunk 4]
```diff
@@ -1,126 +1,229 @@
...
+	for i := range pods.Items {
 
-	awaitServices(t, kubeOptions, filter)
+		// Wait all deployed pods to be available and ready
+		k8s.WaitUntilPodAvailable(t, kubeOptions, pods.Items[i].Name, 60, 1*time.Second)
 
-	tunnel := k8s.NewTunnel(kubeOptions, k8s.ResourceTypePod, prometheusPod, 0, 9090)
-	defer tunnel.Close()
-	tunnel.ForwardPort(t)
+		var filter string = "server"
```
Comment (omarahm3) [inline test/infra_test.go:165]: use short assignment instead

Comment (omarahm3) [reply test/infra_test.go:165]: Actually you can use "server" directly no need to use in a variable at all, or move to const

**Review by omarahm3 — CHANGES_REQUESTED** [review]

**Review by omarahm3 — APPROVED** [review]

**Pull request merged**